src/app/onion/learning-object-builder/builder-store.service.ts

```diff
@@ -449,20 +453,25 @@ export class BuilderStore {
 
   /**
    * Updates Url at given index
-   *
+   * Also checks for valid Url and title field input against the supplied regex pattern
    * @param {number} index
    * @param {Url} url
    * @memberof BuilderStore
    */
   private async updateUrl(index: number, url: Url): Promise<any> {
-    if (!url.url.match(/https?:\/\/.+/i)) {
-      url.url = `http://${url.url}`;
-    }
-    this.learningObject.materials.urls[index] = url;
-    await this.saveObject({
-      'materials.urls': this.learningObject.materials.urls
-    });
-    this.learningObjectEvent.next(this.learningObject);
+    var exp = /^(?:http(s)?:\/\/)?[\w.-]+(?:\.[\w\.-]+)+[\w\-\._~:/?#[\]@!\$&'\(\)\*\+,;=.]+$/;
+    if (url.url.match(exp) && url.title !== '') {
+      url.title = `${url.title}`;
```

Comment:
We shouldn't need to wrap these in string template literals if we're not mutating the string in any way.